### PR TITLE
Fixes #23088 - remove dynflow dependency

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: katello.gemspec [hunk 1]
```diff
@@ -28,7 +28,6 @@ Gem::Specification.new do |gem|
 
   gem.add_dependency "rabl"
   gem.add_dependency "foreman-tasks", "~> 0.12"
-  gem.add_dependency "dynflow", "~> 0.8.31"
```
Comment (jturel) [inline katello.gemspec]: why were we locking this anyway?

Comment (jturel) [reply katello.gemspec]: i guess the issue basically explains it :)

Comment (jlsherrill) [reply katello.gemspec]: I'm not 100% sure, it was added here:  https://github.com/Katello/katello/commit/ea51f03c05436c04c5f8a79c5fa8b8fce007864a  
But core did not require it back then, and it does now.

**Comment by jlsherrill** [comment]
Tests passed with this, but rubocop is still failing, will address in a 2nd PR!

File: katello.gemspec (hunk 1 above)
Comment (adamruzicka) [reply katello.gemspec]: From what I recall we needed to get version of Dynflow which had `::Dynflow::Action::Singleton` module in order for the changes to work.

**Pull request merged**